### ISSUE TITLE
tsc build and watch tasks fail when invoked by shift-cmd-b

### ISSUE THREAD

**Issue by dgh1000** [microsoft/vscode#83128]
Got no answers on stack overflow after a week. No comments even.

I'm working on a typescript project. Previously I could invoke the tsc build and watch commands via shift-cmd-B and they worked fine. Now, I get

env: node: no such file or directory

Meanwhile, if I just start and integrated terminal and type 'tsc --watch' it's fine. The only trouble with the latter is that the error messages aren't clickable.

I can't find any explanation of how it invokes the tsc watch and build tasks, such as what shell or environment it's using. They aren't in tasks.json (there is no tasks.json)

**Comment by alexr00** [comment]
If you open the developer tools (command Toggle Developer Tools) and look in the console do you see any related errors?

What version of VS Code are you using? What OS are you on?

**Comment by dgh1000** [comment]
Thanks for your help. I'm using version 1.39.2 on macOS High Sierra. Some errors do appear in the developer's console when I run VSCode, load my project, and try to build. Although I don't know if they are related. I've attached a screen shot. 
![Screenshot 2019-10-23 02 27 19](https://user-images.githubusercontent.com/16109382/67379028-0bb91580-f53d-11e9-9ba5-746efe3ea237.png)

**Comment by alexr00** [comment]
Does a simple task work if you use the Run Task command?
```
        {
            "label": "two",
            "type": "shell",
            "command": "echo two",
            "problemMatcher": []
        },
```

**Comment by dgh1000** [comment]
Yes, that works.

**Comment by alexr00** [comment]
Can you open the dev tools and in the console check the result of `process.env`? It sounds like node is not on your path.

**Comment by dgh1000** [comment]
Hi Alex,
Before I did what you just suggested, you got me thinking about why node
would not be on the path, and I stumbled across some configuration that I
or someone put into my settings.json (I think maybe a Python extension did
this when it noticed I was running a conda environment): the setting for
"inherit the environment in integrated terminal" was false. Changing that
to true fixed it. I know I didn't change this setting consciously.

So this is fixed.

Mike
